### PR TITLE
Always use the namespace in entities switchsets

### PR DESCRIPTION
**Pull request by palourde**
Signed-off-by: Simon Plourde <simon@sensu.io>

## What is this change?

It ensures we always use the entity's namespace within the keepalives switchets. Previously, we would bury `/sensu.io/switchsets/keepalived/whisky.local` while creating a lease for `/sensu.io/switchsets/keepalived/default/whisky.local` (notice the `default` namespace), so the switchsets would never actually be deleted.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/2760

## Does your change need a Changelog entry?

Added!

## Do you need clarification on anything?

I've tested and re-tested this change with multiple scenarios, I believe it's 100% working but I'm not totally familiar with this codebase.

## Were there any complications while making this change?

Nope!

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Nope!

## How did you verify this change?

Manually tested with a 3-nodes cluster, probably a good idea to test against sensu-staging once the benchmarking is done.

### CODE REVIEW

File: backend/liveness/liveness.go [hunk 1]
```diff
@@ -139,7 +139,8 @@ func NewSwitchSet(client *clientv3.Client, name string, dead, alive EventFunc, l
 // Bury buries a live or dead switch. The switch will no longer
 // or callbacks.
 func (t *SwitchSet) Bury(ctx context.Context, key string) error {
-	key = path.Join(t.prefix, key)
+	t.logger.WithFields(logrus.Fields{"key": key}).Debug("buring key")
```
Comment (echlebek) [inline backend/liveness/liveness.go]: s/buring/burying/

**Comment by palourde** [comment]
TODO @palourde : Figure out how to delete the swithsets keys that do not include a namespace, so they don't spam etcd with PUT operations!

**Comment by palourde** [comment]
@echlebek I've realized my change would probably have broke the check TTL feature, so I've done a bit of refactoring so all 3 public methods of the liveness interface accept an _ID_ instead of the full key path. Also, I modified the keepalived logic so if it receives a "bad key", it buries the key directly instead of letting it trigger a new event in the watcher.

I've tested keepalives & check TTLs, everything seems to be working now!

File: backend/keepalived/keepalived.go [hunk 2]
```diff
@@ -328,23 +329,29 @@ func (k *Keepalived) alive(key string, prev liveness.State, leader bool) bool {
...
+		// events to the watcher.
 		lager.Error(err)
-		return false
+		return true
 	}
 
 	lager = lager.WithFields(logrus.Fields{"entity": name, "namespace": namespace})
 	lager.Warn("keepalive timed out")
 
-	if !leader {
```
Comment (echlebek) [inline backend/keepalived/keepalived.go]: Why has this been removed? This prevents the non-leader nodes from making unnecessary requests.

Comment (palourde) [reply backend/keepalived/keepalived.go]: The diff is confusing but it has only been moved up, on line 336, so we don't return `true`, and therefore bury the key if it's invalid and we are not the leader. Makes sense?

Comment (echlebek) [reply backend/keepalived/keepalived.go]: I see! That makes sense, but wondering if it should happen after the logger gets its additional context (namespace, etc)

Comment (palourde) [reply backend/keepalived/keepalived.go]: Ah yes, I can move things around so we add the entity name & namespace before logging everything and then check the error.

**Pull request merged**